core/corehttp/gateway_handler.go

```diff
@@ -92,16 +92,24 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 
 	urlPath := r.URL.Path
 
+	// If the gateway is behind a reverse proxy and mounted at a sub-path,
+	// the prefix header can be set to signal this sub-path.
+	// It will be prepended to links in directory listings and the index.html redirect.
+	prefix := ""
+	if prefixHdr := r.Header["X-Ipfs-Gateway-Prefix"]; len(prefixHdr) > 0 {
+		log.Debugf("X-Ipfs-Gateway-Prefix: %s", prefixHdr[0])
+		prefix = prefixHdr[0]
+	}
```

Comment:
I'm unsetting this header for all other vhosts in nginx. The prefix "only" affects rendering of URLs (and redirects), do you think that opens up attack surface in other setups? One thing I should definitely make sure is that you can pass full URLs, but only what's expected (paths). An attacker must not be able to trick a server into arbitrary redirects.